### ISSUE TITLE
fit_members needs a new_data option

### ISSUE THREAD

**Issue by py9mrg** [tidymodels/stacks#104]
## Feature

When building a stack it is natural that all the work is done on the training set, so that we can check accuracy on the test set. But after all that, we often want the very last step to train the model on the full data.

For example, if we were building a random forest, we might do something like:

``` r
# imagine we already have a workflow, data_split and tuning done, we'd want to check what our accuracy might be using this model

rf_wf |>
  finalize_workflow(whatever you choose) |>
  last_fit(data_split) |>
  pull(.metrics)

# now we might want to train the model on the full data, ready for entirely new data, rather than limit training to the training set

rf_model <- rf_wf |>
  finalize_workflow(whatever you choose) |>
  fit(full_data)
```
This is possible thanks to `fit` having a `new_data` argument. However, with `fit_members` there isn't this possibility so the final model has to remain trained only on the training set.

**Comment by topepo** [comment]
My thoughts from tidymodels/tune#470

> A stacked ensemble uses out-of-sample predictions to create the ensemble. If you use the test set, you have no method for determining if the ensemble is working. Also, you would be evaluating numerous models on the test set and that's problematic. 
>
> I don't think that this is a great idea so we won't be making any changes. 
>
> Perhaps it would be better to use a validation set as the input into the data stack? You can use `validation_split()` to achieve this. 

Reading this issue, I can see how you'd like to use the entire training set. 

I get that, but I think that we should err on the side of not doing that. Depending on the size of your training and test sets, the results that you get from using the training set may not translate to what you would get when you combine the two sets. 

Although it is inefficient, I would suggest taking the models that you are interested in, combining the training and test sets, the use someone like cross-validation to generate the out-of-sample results to give to the stack. 

In doing this you would get a much better ensemble since it would use all of your data to estimate the ensemble weights. If you were to use the test set, it would only be based on that smaller data partition. 

Either way, I'd try to convince you that it is not a great idea since you would be flying blind; there's no way to evaluate the ensemble now.

**Comment by py9mrg** [comment]
No worries. I probably need to think a bit harder about exactly what's going on under the hood of stacks as I naively thought it would be fine to mimic a non-stacked approach. Indeed I thought about your idea but was so worried about data leakage I wanted to keep the test set entirely separate. Probably your idea of a validation set over on the other issue is a good way to go. I'll close both, and thank for the advice.
